File: Week2/50.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        def quick(N):
            cur=x
            res=1
            s=bin(N)[2:][::-1]
            for i in s:
                if i=="1":
                    res*=cur
                cur*=cur
            return res
        return quick(n) if n>0 else 1.0/quick(n*-1)

File: Week2/50.java
class Solution {
    public double myPow(double x, long n) {
    if(n<0) return 1/myPow(x,-n);
    if(n==0) return 1;
    if(n%2==0) return Math.pow(myPow(x,n/2),2);
    return x*Math.pow(myPow(x,n/2),2);

    }
}